src/rgw/rgw_lc.cc

```diff
@@ -92,6 +96,19 @@ int RGWLifecycleConfiguration::check_and_add_rule(LCRule *rule)
   return 0;
 }
 
+bool RGWLifecycleConfiguration::has_same_action(const lc_op& first, const lc_op& second) {
+  if ((first.expiration > 0 || first.expiration_date != boost::none) && 
```

Comment:
I think a better solution is to add an "empty()" or "valid()" method to LCExpiration that checks date and days.  That way all the checks are simple for all the rule types.  If "empty()" is used, then all the checks above can stay the same.

This also means that the code checking validity doesn't need to change when we add new types in the future.